client/lib/support/support-user/test/localforage-bypass.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import { spy, stub } from 'sinon';
+
+/**
+ * Internal dependencies
+ */
+import localForageBypass from '../localforage-bypass';
+
+describe( 'localforage-bypass', () => {
+	const localForage = localForageBypass;
+	let db = null;
+
+	beforeEach( ( done ) => {
+		localForage.ready = () => Promise.resolve();
+		localForage._initStorage( {} )
+			.then( () => {
+				db = localForage._dbInfo.db;
+				done();
+			} )
+			.catch( done );
+	} );
+
+	describe( 'keys', () => {
+		it( 'should list all keys', ( done ) => {
+			db.one = 1;
+			db.two = 2;
+			localForage.keys()
+				.then( ( keys ) => {
+					expect( keys ).to.have.length( 2 );
+					expect( keys ).to.deep.equal( [ 'one', 'two' ] );
+					done();
+				} )
+				.catch( done );
+		} );
+
+		it( 'should be empty when initialized', ( done ) => {
+			localForage.keys()
+				.then( ( keys ) => {
+					expect( keys ).to.have.length( 0 );
+					done();
+				} )
+				.catch( done );
+		} );
+	} );
+
+	describe( 'length', () => {
+		const length = () => localForage.length();
+		const expectLength = ( expected ) => {
+			return () => {
+				return length()
+					.then( ( l ) => expect( l ).to.equal( expected ) );
+			}
+		};
+		const addItem = ( key, value ) => {
+			return () => localForage.setItem( key, value )
+		};
+
+		it( 'should be zero when initialized', ( done ) => {
+			expectLength( 0 )()
```

Comment:
Mocha has built-in support for promises, so you don't need to call `done` yourself:

https://mochajs.org/#working-with-promises